src/stories/containers/recognized-delegates/useRecognizedDelegates.mvvm.tsx

```diff
@@ -110,6 +169,507 @@ const useRecognizedDelegates = (delegates: DelegatesDto) => {
+    const result: InnerTableRow[] = [];
+    if (!wallets) {
+      return result;
+    }
+
+    if (hasExpenses(true)) {
+      result.push({
+        items: [
+          {
+            column: breakdownColumns[0],
+            value: 'Headcount Expenses',
+          },
+          {
+            column: breakdownColumns[1],
+            value: hasGroups ? '' : 'Headcount Expenses',
+          },
+        ],
+        type: 'section',
+      });
+
+      result.push(
+        ...getBreakdownItems(currentWallet?.budgetStatementLineItem?.filter((item) => item.headcountExpense))
+      );
+
+      result.push(
+        ...getBreakdownItems(
+          [
+            getLineItemsSubtotal(
+              currentWallet?.budgetStatementLineItem?.filter((item) => item.headcountExpense),
+              'Subtotal'
+            ),
+          ],
+          'subTotal'
+        )
+      );
+    }
+
+    if (hasExpenses(false)) {
+      result.push({
+        items: [
+          {
+            column: breakdownColumns[0],
+            value: 'Non-Headcount Expenses',
+          },
+          {
+            column: breakdownColumns[1],
+            value: hasGroups ? '' : 'Non-Headcount Expenses',
+          },
+        ],
+        type: 'section',
+      });
+
+      const headcountExpenseItems = getBreakdownItems(
+        currentWallet?.budgetStatementLineItem?.filter((item) => !item.headcountExpense)
+      );
+
+      result.push(...headcountExpenseItems);
+
+      result.push(
+        ...getBreakdownItems(
+          [
+            getLineItemsSubtotal(
+              currentWallet?.budgetStatementLineItem?.filter((item) => !item.headcountExpense),
+              'Subtotal'
+            ),
+          ],
+          'subTotal'
+        )
+      );
+    }
+
+    if (result.length > 0) {
+      result.push({
+        type: 'total',
+        items: [
+          {
+            column: breakdownColumns[0],
+            value: hasGroups ? 'Total' : '',
+          },
+          {
+            column: breakdownColumns[1],
+            value: hasGroups ? '' : 'Total',
+          },
+          {
+            column: breakdownColumns[2],
+            value: getWalletForecast(currentWallet),
+          },
+          {
+            column: breakdownColumns[3],
+            value: getWalletActual(currentWallet),
+          },
+          {
+            column: breakdownColumns[4],
+            value: getWalletDifference(currentWallet),
+          },
+          {
+            column: breakdownColumns[5],
+            value: '',
+          },
+          {
+            column: breakdownColumns[6],
+            value: getWalletPayment(currentWallet),
+          },
+        ],
+      });
+    }
+
+    return result;
+  }, [
+    breakdownColumns,
+    currentWallet,
+    getBreakdownItems,
+    getLineItemsSubtotal,
+    getWalletActual,
+    getWalletDifference,
+    getWalletForecast,
+    getWalletPayment,
+    hasExpenses,
+    hasGroups,
+    wallets,
+  ]);
```

Comment:
All this can be moved to delegates-actuals or delegates-forecasts as they're managing its functionality in its own hook

Reply:
you are right, done